cli/command/network/list_test.go

```diff
@@ -41,23 +40,53 @@ func TestNetworkListErrors(t *testing.T) {
 	}
 }
 
-func TestNetworkListWithFlags(t *testing.T) {
-	expectedOpts := types.NetworkListOptions{
-		Filters: filters.NewArgs(filters.Arg("image.name", "ubuntu")),
-	}
+func TestNetworkList(t *testing.T) {
+	testCases := []struct {
+		doc             string
+		networkListFunc func(ctx context.Context, options types.NetworkListOptions) ([]types.NetworkResource, error)
+		flags           map[string]string
+		golden          string
+	}{
+		{
+			doc: "network list with flags",
+			flags: map[string]string{
+				"filter": "image.name=ubuntu",
+			},
+			golden: "network-list.golden",
+			networkListFunc: func(ctx context.Context, options types.NetworkListOptions) ([]types.NetworkResource, error) {
+				expectedOpts := types.NetworkListOptions{
+					Filters: filters.NewArgs(filters.Arg("image.name", "ubuntu")),
+				}
+				assert.Check(t, is.DeepEqual(expectedOpts, options, cmp.AllowUnexported(filters.Args{})))
 
-	cli := test.NewFakeCli(&fakeClient{
-		networkListFunc: func(ctx context.Context, options types.NetworkListOptions) ([]types.NetworkResource, error) {
-			assert.Check(t, is.DeepEqual(expectedOpts, options, cmp.AllowUnexported(filters.Args{})))
-			return []types.NetworkResource{*NetworkResource(NetworkResourceID("123454321"),
-				NetworkResourceName("network_1"),
-				NetworkResourceDriver("09.7.01"),
-				NetworkResourceScope("global"))}, nil
+				return []types.NetworkResource{*NetworkResource(NetworkResourceID("123454321"),
+					NetworkResourceName("network_1"),
+					NetworkResourceDriver("09.7.01"),
+					NetworkResourceScope("global"))}, nil
+			},
 		},
-	})
-	cmd := newListCommand(cli)
+		{
+			doc: "network list sort order",
+			flags: map[string]string{
+				"format": "{{ .Name }}",
+			},
+			golden: "network-list-sort.golden",
+			networkListFunc: func(ctx context.Context, options types.NetworkListOptions) ([]types.NetworkResource, error) {
+				return []types.NetworkResource{
+					*NetworkResource(NetworkResourceName("network-2-foo")),
+					*NetworkResource(NetworkResourceName("network-1-foo")),
+					*NetworkResource(NetworkResourceName("network-10-foo"))}, nil
+			},
+		},
+	}
 
-	cmd.Flags().Set("filter", "image.name=ubuntu")
-	assert.NilError(t, cmd.Execute())
-	golden.Assert(t, strings.TrimSpace(cli.OutBuffer().String()), "network-list.golden")
+	for _, tc := range testCases {
+		cli := test.NewFakeCli(&fakeClient{networkListFunc: tc.networkListFunc})
```

Comment:
Looks like we have everything in place to use subtests here (only change two lines);

```patch
diff --git a/cli/command/network/list_test.go b/cli/command/network/list_test.go
index 55d0b109..2f76e517 100644
--- a/cli/command/network/list_test.go
+++ b/cli/command/network/list_test.go
@@ -81,12 +81,14 @@ func TestNetworkList(t *testing.T) {
        }
 
        for _, tc := range testCases {
-               cli := test.NewFakeCli(&fakeClient{networkListFunc: tc.networkListFunc})
-               cmd := newListCommand(cli)
-               for key, value := range tc.flags {
-                       cmd.Flags().Set(key, value)
-               }
-               assert.NilError(t, cmd.Execute())
-               golden.Assert(t, cli.OutBuffer().String(), tc.golden)
+               t.Run(tc.doc, func(t *testing.T) {
+                       cli := test.NewFakeCli(&fakeClient{networkListFunc: tc.networkListFunc})
+                       cmd := newListCommand(cli)
+                       for key, value := range tc.flags {
+                               cmd.Flags().Set(key, value)
+                       }
+                       assert.NilError(t, cmd.Execute())
+                       golden.Assert(t, cli.OutBuffer().String(), tc.golden)
+               })
        }
 }
```

Reply:
Thank you for the review. Fixed.